public/app/plugins/datasource/prometheus/components/PromQueryField.tsx

```diff
@@ -302,11 +302,13 @@ class PromQueryField extends React.PureComponent<PromQueryFieldProps, PromQueryF
     return (
       <>
         <div className="gf-form-inline gf-form-inline--nowrap flex-grow-1">
-          <div className="gf-form flex-shrink-0">
-            <ButtonCascader options={metricsOptions} disabled={buttonDisabled} onChange={this.onChangeMetrics}>
-              {chooserText}
-            </ButtonCascader>
-          </div>
+          {this.languageProvider && (
```

Comment:
Also lets make the type of `this.languageProvider` property optional here.

Reply:
What do you mean by this? It **is** optional and that's why the ButtonCascader will render or not, depending on whether it is set?

Reply:
The same thing as in datasource, it is defined as `languageProvider: PromQlLanguageProvider;` and I guess it should be `languageProvider?: PromQlLanguageProvider;`

Reply:
Ohh, so the same thing that you also noted above? Yeah, that's already addressed. :)